tests/ui/mod_lib/main.rs

```diff
@@ -0,0 +1,8 @@
+#![allow(unused)]
+#![warn(clippy::mod_lib)]
+
+mod lib;
```

Comment:
I think a `mod lib { }` should suffice here, which obviates the need for a secondary file.